tidb-binlog/pump_client/client.go

```diff
@@ -335,6 +321,8 @@ func (c *PumpsClient) backoffWriteBinlog(req *pb.WriteBinlogReq, binlogType pb.B
 	var resp *pb.WriteBinlogResp
 	// send binlog to unavaliable pumps to retry again.
 	for _, pump := range unAvaliablePumps {
+		pump.ResetGrpcClient()
```

Comment:
why add it, if it is in `unAvaliablePumps` , it must call ResetGrpcClient at `setPumpAvaliable` or dont create grpc client

Reply:
because we will send binlog to unavailable pump in backoffWritebinlog function, will create grpc connection, this connection may be unavailable, so I reset the connection here.